### ISSUE TITLE
memcpy transform to sw with unaligned buffers

### ISSUE THREAD

**Issue by ghost** [upmem/llvm-project#16]
[test_aligment.zip](https://github.com/upmem/llvm-project/files/3910655/test_aligment.zip)

```
     2: 00000020     1 OBJECT  LOCAL  DEFAULT    3 c
     8: 00000021     8 OBJECT  GLOBAL DEFAULT    3 b
     7: 00000030     8 OBJECT  GLOBAL DEFAULT    3 a
```

```
80000088 bar:
...
80000098: 00 00 12 00 63 61 00 00       move r2, 0x21
800000a0: 00 00 02 44 08 7c 00 00       sw r2, 0x0, r1
```

as `c` makes `b` not aligned on `8` bytes, the transformation of memcpy to sw is not legal and produces a `memory fault` at runtime.

**Comment by ghost** [comment]
Seen initially from tests in supertest:
```
4/11/2/3/t2.c		
4/11/3/1/t1.c
C99/7/21/2/3/t2.c		
￼C99/7/21/3/1/t1.c
```

**Comment by ghost** [comment]
fix by upmem/llvm-project@c8c7ca7623a
